velox/functions/prestosql/aggregates/PrestoHasher.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/vector/DecodedVector.h"
+
+namespace facebook::velox::aggregate {
+
+class PrestoHasher {
+ public:
+  explicit PrestoHasher(
+      const BaseVector& vector,
+      const SelectivityVector& rows) {
+    vector_ = std::make_shared<DecodedVector>(vector, rows);
+    createChildren();
+  }
+
+  /// Computes a hash identical to the one used by Presto checksum. The
+  /// algorithm is based on XXHash64. Takes SelectivityVector of rows to be
+  /// hashed , These rows must be a subset of the rows passed to the
+  /// constructor. The resultant hashes are saved in BufferPtr hashes, which
+  /// must have as much capacity as sizeof(int64_t) * rows.end().
+  void hash(const SelectivityVector& rows, BufferPtr& hashes);
```

Comment:
A question about the API.

I'm seeing that PrestoHasher instance can be used to compute hashes for a single vector. It can be used to compute multiple sets of hashes for different subset of rows in a given vector. 

Assuming PrestoHasher is intended to be used as a building block for checksum aggregate function, it would be nice to allow for a single instance to be used for hashing multiple vectors of the same type.

PrestoHasher contains DecodedVector's which allocate memory. Reusing PrestoHasher instances would allow for reusing DecodedVector's and the memory they allocate. 

Hence, I envision a slightly different API where PrestoHasher is created using a TypePtr that specified the type of vectors it is capable of hashing and provides a hash() function that takes a vector, a SelectivityVector, and an output buffer of hashes. This way, checksum aggregate function could use a single instance of PrestoHasher to hash all batches of input. 

```
PrestoHasher(TypePtr type);

void hash(const BaseVector& vector, const SelectivityVector& rows, BufferPtr& hashes);
```